### PR TITLE
Make atom names unique if needed

### PR DESCRIPTION
**Pull request by raimis**
- [x] Make atom names unique if needed
  - [x] Implement `makeAtomNamesUnique`
- [x] Update tests

~~Depend on #788 and #792~~

Depend on https://github.com/Acellera/htmd-data/pull/11

### CODE REVIEW

**Comment by j3mdamas** [comment]
I'm puzzled by this. What does this new function is supposed to do
different from the canolicalizeNames function?

File: htmd/parameterization/fftype.py [hunk 1]
```diff
@@ -125,124 +142,114 @@ def fftype(mol, rtfFile=None, prmFile=None, method='GAFF2', acCharges=None, tmpD
                     raise RuntimeError('"antechamber" failed')
 
                 # Run parmchk2
-                returncode = subprocess.call([parmchk2_binary,
-                                              '-f', 'prepi',
-                                              '-s', atomtype,
-                                              '-i', 'mol.prepi',
-                                              '-o', 'mol.frcmod',
-                                              '-a', 'Y'], cwd=tmpdir)
+                cmd = ['parmchk2',
```
Comment (stefdoerr) [inline htmd/parameterization/fftype.py]: Why this? I think it's better to check for the binary as it was before.

**Comment by stefdoerr** [comment]
So what is the difference with canonicalizeAtomNames? That it maintains existing atom names? How did the previous fail? Would be useful I think to have a description in the PR

File: htmd/parameterization/cli.py [hunk 2]
```diff
@@ -73,17 +73,22 @@ def getArgumentParser():
...
 
-    # Check if each atom name is unique
+    # Make atom names unique if needed
     if np.unique(mol.name).size != mol.numAtoms:
-        raise RuntimeError('The atom names in {} has to be unique!'.format(args.filename))
+        logger.warning('Atom names in the molecule are not unique!')
+        new_mol = makeAtomNamesUnique(mol)
+        for i, (old_name, new_name) in enumerate(zip(mol.name, new_mol.name)):
+            if old_name != new_name:
+                logger.warning('Rename atom {:3d}: {:4s} --> {:4s}'.format(i, old_name, new_name))
```
Comment (j3mdamas) [inline htmd/parameterization/cli.py]:
```suggestion
                logger.warning('Renamed atom {:3d}: {:4s} --> {:4s}'.format(i, old_name, new_name))
```

File: htmd/parameterization/util.py [hunk 3]
```diff
@@ -296,6 +296,77 @@ def guessBondType(mol):
...
+    >>> new_mol = makeAtomNamesUnique(mol)
+    >>> assert new_mol is not mol
+    >>> new_mol.name
+    array(['1A', '1A0', 'A1B1', 'A1B2'], dtype=object)
+    """
+
+    from htmd.molecule.molecule import Molecule
+
+    if not isinstance(mol, Molecule):
+        raise TypeError('"mol" has to be instance of {}'.format(Molecule))
```
Comment (j3mdamas) [inline htmd/parameterization/util.py]:
```suggestion
        raise TypeError('"mol" has to be an instance of {}'.format(Molecule))
```

File: htmd/parameterization/fftype.py (hunk 1 above)
Comment (raimis) [reply htmd/parameterization/fftype.py]: `subprocess.call` it does implicitly

Comment (stefdoerr) [reply htmd/parameterization/fftype.py]: It was more to give a more sensible error message to the user but whatever. It's not supposed to be missing anyway

**Pull request merged**